Gadm resolver Feature, rebased to main

ccjoel opened the pull request:
Gadm Resolver UI Feature. Ready to merge.

GoogleSheets commented:
### Description

- modified convert_bbox to handle new bounding box format with two points: 
`--bbox='[[33.512234, 2.719907], [49.98171,16.501768]]'`

- performs a simple length check on the bbox array and converts the new format to old

```
def convert_bbox(self, bb):
        """
        Convert WGS84 coordinate system to Web Mercator
        Initial bbox is in format [xmin, ymin, xmax, ymax].
        New bbox box format is [[xmin, ymin], [xmax, ymax]]
        x is longitude, y is latitude.
        Output is Web Mercator min/max points for a bounding box.
        """

        in_proj = Proj(init='epsg:4326')
        out_proj = Proj(init='epsg:3857')

        if len(bb) == 2:
            # Convert new bbox format to old bbox format.
            bb = bb[0] + bb[1]
        
        min_pt = transform(in_proj, out_proj, bb[0], bb[1])
        max_pt = transform(in_proj, out_proj, bb[2], bb[3])
     
        return min_pt, max_pt  

```

ccjoel committed:
Mock (in code) Gadm resolution step for Dojo dataset registration.

ccjoel committed:
CHeckpoint. Gadmresolver in the data transform panel on dataset registration.
Mocked job results so far.

ccjoel committed:
Some ntermediary cleanup.

ccjoel committed:
UI v2 implemented for gadm resolution within the transform step of data
transformation. Integration with backend (currently mock data) and changes from
feedback pending.

ccjoel committed:
CHanged gadm resolver table shadow to a thin border, and the table heading font
size was reduced.

ccjoel committed:
Gadm resolver UI making real rq job request to check for the data. Need to wire up the 'save' portion, clean up, and see what well do to implement the backend work as it is currently mocked.

ccjoel committed:
Changed schema of gadm detection, among other improvements.

ccjoel committed:
Almost done. Saved gadm resolution set by user. Updated stories. Need to change
owner of some state and will be in good shape.

ccjoel committed:
Gadm Overrides are reloaded from DataTransformation to GadmResolver when opening
and closing the panel.

ccjoel committed:
Removed duplicated required check for data transform review step- leftover from a merge.

Sorrento110 committed:
GADM resolver job updated.

ccjoel committed:
Integrated GadmResolver with backend. Added some other exception handling for
when jobs finish too quicly with either success data on initial call, or a full
backend job_error (irrecoverable; where we display a stacktrace dialog for now).

Sorrento110 committed:
Added in job to get all available gadm strings.

ccjoel committed:
Fetching list of all gadm countries in order to properly populate autocomplete.

ccjoel committed:
Cleaned up old code, old stories for it.

ccjoel committed:
Lint fixes.

ccjoel committed:
Handling lint ptoblems, removing stacktrace from errors from starting a elwood job.

Sorrento110 committed:
Look for the geo_type related to the admin_level.

ccjoel committed:
Adapted cypress dataset register tests to work with new data transform step
required operations.

ccjoel committed:
CHanged gadm overrides schema to allow data and to allow arbitrary child keys.

ccjoel committed:
Changing order of overrides and gadm key for gadm overrides.

ccjoel committed:
Using gadm overrides on the rqworker tasks side to pass in to elwood run_elwood job.

ccjoel committed:
Forgot to pass a paramater, fixing. FIxing regression that would occur only on
local dev.

ccjoel committed:
Configuring logger for two processors in order to inspect the info logs that are
already in place within gadm,elwood processors (rqworker).

ccjoel committed:
Ensure we update dataset annotations in react state so that run_elwood uses gadm
user overrides. this is on data transformation component.

ccjoel committed:
Addressed Review Feedback.

ccjoel committed:
Fixing typoe after review fixes.

ccjoel committed:
New lint fixes after rebasing.

ccjoel commented:
Ignore the dataset register test failing, seems to be a fluke (to be addressed some other time).
